File: 351003/Naskevich/src/models/tweet.py
from dataclasses import dataclass, field
from datetime import datetime, timezone


@dataclass(kw_only=True)
class Tweet:
    id: int = field(default=0, init=False)
    editor_id: int
    title: str
    content: str
    created: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    modified: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    markers: list = field(default_factory=list)

File: 351003/Mashevskiy/Lab2/src/main/java/by/tracker/rest_api/entity/Tweet.java
package by.tracker.rest_api.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "tbl_tweet")
public class Tweet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String content;
    private LocalDateTime created;
    private LocalDateTime modified;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    private Creator creator;

    @ManyToMany
    @JoinTable(
            name = "tbl_tweet_marker",
            joinColumns = @JoinColumn(name = "tweet_id"),
            inverseJoinColumns = @JoinColumn(name = "marker_id")
    )
    private List<Marker> markers = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        created = LocalDateTime.now();
        modified = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        modified = LocalDateTime.now();
    }
}